programs/stake/src/stake_state.rs

```diff
@@ -105,11 +106,25 @@ fn redelegate(
 ) -> Result<(), StakeError> {
     // If stake is currently active:
     if stake.stake(clock.epoch, Some(stake_history)) != 0 {
+        let stake_lamports_ok = if invoke_context
+            .feature_set
+            .is_active(&feature_set::stake_redelegate_instruction::id())
+        {
+            // When a stake account is redelegated, the lamports from the source stake account are
+            // transferred to a new stake account. Do not permit the deactivation of the source stake
+            // account to be rescinded, by more generally requiring the delegation be configured
+            // with the expected amount of stake lamports before rescinding.
```

Comment:
heh true. and I agree, it's weird but seems just fine. the original deactivation is rescinded because the original stake has been restored, and the "new" stake account is now activating normally.